src/averageAges.js

```diff
@@ -14,12 +14,25 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateMenAverageAge(people, century = 0) {
+  function withCentury() {
+    const foundation = people.filter(person => person.sex === 'm'
+    && Math.ceil(person.died / 100) === century);
+    const finallyFunc = foundation.reduce((sum, pers) =>
+      sum + pers.died - pers.born, 0);
+
+    return finallyFunc / foundation.length;
+  }
+
+  function withoutCentury() {
+    const foundation = people.filter(person => person.sex === 'm');
+    const finallyFunc = foundation.reduce((sum, person) =>
+      sum + person.died - person.born, 0);
+
+    return finallyFunc / foundation.length;
+  }
+
+  return century > 0 ? withCentury() : withoutCentury();
```

Comment:
let's simplify your solution
at first you should create variable where you should filter people array depending on the condition
also use more meaningful names
then create separate function outside, that will calculate average age, and use it here and in the next
function